### DISCUSSION TITLE
About quantiles and aggretation

### DISCUSSION THREAD

**Discussion by pedrolamarao** [open-telemetry/opentelemetry-specification#3118] [General]
Hello, all! 
Thank you very much for your hard work on OpenTelemetry!

I have been looking into applying OpenTelemetry in my products as a replacement for my own custom telemetry code.
We are looking into applying OpenTelemetry interface across our components and rewriting our infrastructure with a custom collector or exporter.
One of the features we depend upon is online accumulation of statistics information such as quantiles.

My understanding of the ticket below is that a "summary" instrument was experimented on but ultimately removed based on concerns over summability.
In that ticket, many participants assert that quantile information is not summable.
https://github.com/open-telemetry/opentelemetry-proto/pull/199

We have for some time now been using TDigests for online accumulation of statistics.
TDigests are summable.
They support very convenient use cases such as accumulating digests per time window, serializing for storage, and later aggregating them for visualization of specified time periods.
I suppose that in OpenTelemetry parlance these would be "deltas".
https://github.com/tdunning/t-digest

Is it possible to extend OpenTelemetry somehow to support new instruments such as a TDigest?

**Comment by tdunning** [comment]
To add a little flavor there, t-digests are a distributional sketch that particularly focusses on maintaining high accuracy for quantiles near the tails. Typical accuracy is ppm for percentiles below 5% and above 95% with strictly bounded size.

As an additional property, t-digests can be merged while maintaining accuracy. They are commonly used in many applications.

I am happy to answer any questions on this.
